### PR TITLE
Support right-clicking on iPadOS

### PR DESCRIPTION
**Pull request by moffatman**
iPadOS 13.4 introduced improved cursor support, including native support for more pointer "buttons" than the default. There is also a change needed to the framework to avoid adding the default button to iPad right-clicks (link tba). 

flutter/flutter#55809

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @gaaclarke

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m
Comment (gaaclarke) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:20]: I don't think you want conditional compilation here, you want runtime checks like you have below.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m
Comment (gaaclarke) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:22]: How come this is asking for iOS 15 when the implementation check is for iOS 13.4?

**Review by gaaclarke — COMMENTED** [review]
Outstanding contribution, thanks!  I just have a few nits/questions, otherwise LGTM.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m
Comment (moffatman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:20]: I'm not sure this will compile on Xcode 12 without this conditional, since supportsPointerInteraction was only introduced in the Xcode 13 beta.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m
Comment (moffatman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:22]: While the pointer/trackpad features on iPad came out in iPadOS 13.4, synthesizing those events in XCTest wasn't supported until this year's new Xcode/iPadOS beta releases.

Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:22]: You aren't synthesizing any events here though, right?  At the least please add a comment, preferably one that points to the API that is being used that requires ios 15.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m
Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:20]: If you want to do that use `#if (defined (__IPHONE_15_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_15_0)` then.  We don't just care that the SDK has it, but that the target platform has it.

Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m:20]: A comment wouldn't hurt, a runtime check with `respondsToSelector` would be a bit more apparent and easier to maintain, but a comment will be good enough.

**Review by gaaclarke — APPROVED** [review]
I'm going to approve this and trust you to put the comments in the code so someone else can just hit the merge button.  I'll be out of office for awhile and don't want you to get blocked or have to go through review again.

**Comment by gaaclarke** [comment]
Or better yet, I have a spare moment.  I'll land this and just do the quick clean up in a follow up PR so that you're off the hook.  Thanks for the contribution.

**Pull request merged**
